pkg/plugins/golang/v4/scaffolds/internal/templates/readme.go

```diff
@@ -71,9 +71,9 @@ const readmeFileTemplate = `# {{ .ProjectName }}
 
 ### Prerequisites
 - go version v1.22.0+
-- docker version 17.03+.
-- kubectl version v1.11.3+.
-- Access to a Kubernetes v1.11.3+ cluster.
+- a container tool (Docker, Podman, Buildah, etc)
```

Comment:
The issue is that we can't claim full support for all container tools since our testing is limited to Docker. To support additional tools, we would need to test across each of them as well, which would be a considerable burden for us as maintainers. Don't you think?

Reply:
So, my though here was where the requirement appears, which is the **Getting Started** section of the scaffolded README. In this section, we just need to build and push a container image to run the steps described. So should be ok for not test all options here (I tested with Podman, but not with Buildah, for example)

we also have a comment about not testing all options today
https://github.com/kubernetes-sigs/kubebuilder/blob/5f8342e45517e84a8cc8dd4eff825fe340477111/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go#L86-L90

But for other parts of the project, we can still say docker is still required, and no need for changing how is tested (for example, if a section about multi-arch release is added to scaffolded README, `docker-buildx` command would be necessary and then, docker would be necessary requirement for that section)

what you think?

Reply:
Yes, we added a comment to make clear that it **Might** work with other tools. However, it is **not** tested, officially supported.

Reply:
For example, if we allow the tests run here it will broke all our CI as well.

Reply:
agree with waiting for a major update to merge this one

Reply:
So, could you please track an issue with the request, and we can create a go/v5 milestone to add this one there?
When we have enough request changes, we can create a new version and do this one.

Also, could you please link this PR there as a reference and close it?

Reply:
will do

Reply:
created https://github.com/kubernetes-sigs/kubebuilder/issues/4293